staging/src/k8s.io/apiserver/pkg/util/flowcontrol/filter-config.go

```diff
@@ -0,0 +1,349 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	fcboot "k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap"
+	"k8s.io/apiserver/pkg/util/apihelpers"
+	fq "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing"
+	"k8s.io/apiserver/pkg/util/flowcontrol/metrics"
+	"k8s.io/apiserver/pkg/util/shufflesharding"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	rmtypesv1a1 "k8s.io/api/flowcontrol/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+// initializeConfigController sets up the controller that processes
+// config API objects.
+func (reqMgr *requestManager) initializeConfigController(informerFactory kubeinformers.SharedInformerFactory) {
+	reqMgr.configQueue = workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "req_mgmt_config_queue")
+	fci := informerFactory.Flowcontrol().V1alpha1()
+	pli := fci.PriorityLevelConfigurations()
+	fsi := fci.FlowSchemas()
+	reqMgr.plLister = pli.Lister()
+	reqMgr.plInformerSynced = pli.Informer().HasSynced
+	reqMgr.fsLister = fsi.Lister()
+	reqMgr.fsInformerSynced = fsi.Informer().HasSynced
+	pli.Informer().AddEventHandler(reqMgr)
+	fsi.Informer().AddEventHandler(reqMgr)
+}
+
+func (reqMgr *requestManager) triggerReload() {
+	klog.V(7).Info("triggered request-management system reloading")
+	reqMgr.configQueue.Add(0)
```

Comment:
Is this deliberately set up such that changing anything results in rescanning everything?

It's more common to use a key such that changing a thing results in only scanning that thing. That may well not apply here, so a comment explaining that this is deliberate may be in order.

Reply:
Yes, it is deliberate.  The code is a bit simpler to write this way, and I expect configuration changes to never happen in the steady state.  So I thought I would start with simple code.

Reply:
I added a comment at the top of the file.